lxd/daemon_images.go

```diff
@@ -252,6 +254,23 @@ func ImageDownload(r *http.Request, s *state.State, op *operations.Operation, ar
 		ctxMap = logger.Ctx{"fingerprint": info.Fingerprint}
 		logger.Debug("Image already exists in the DB", ctxMap)
 
+		// If the image already exists, is cached and that it is
+		// requested to be downloaded from an explicit `image copy` operation, then disable its `cache` parameter
+		// so that it won't be candidate for auto removal.
+		if imgInfo.Cached && args.DownloadFromExplicitCopy && len(args.TargetAliasesNames) > 0 {
```

Comment:
Why does it need `len(args.TargetAliasesNames) > 0 ` part?

Reply:
can we do this as part of the transaction further down?

Reply:
A misunderstanding on my side.. There is not a real reason for why it is here. Removing it.

Reply:
fixed